### ISSUE TITLE
Getting stuck at sysroot_scripts

### ISSUE THREAD

**Issue by smblogger** [Alex313031/Thorium-Win#319]
**System Details**
 - OS: Windows 10
 - Thorium Version Latest

**Problem**
Build simply gets stuck and doesn't go any further than attempting these. Those files never download. And yes indexing and antivirus are turned off for my /src folder. 

build/linux/sysroot_scripts/install-sysroot.py --arch=amd64 &&
build/linux/sysroot_scripts/install-sysroot.py --arch=arm64 &&

**Additional Notes**
Also had issues with Git Bash not finding python3 even with the c:/depot_tools in the path, but I hard coded the python.exe into the verify script and was able to get past that part.  Still building my first draft.  Any reason you went from the batch files to the shell files?  Just wondering.  Seems the batch would have less issues with paths, etc, on windows systems.

**Comment by gz83** [comment]
Please follow the following documentation to set up your device

https://chromium.googlesource.com/chromium/src/+/main/docs/windows_build_instructions.md

https://github.com/Alex313031/thorium/blob/main/docs/BUILDING_WIN.md

At the same time, please temporarily merge my PR manually to make sure the python files work properly.

https://github.com/Alex313031/thorium/pull/886

Also, please use the python files in //thorium/win_scripts to build the windows version.

trunk.py+version.py+setup.py+Manually set the appropriate args parameters+build_win.py

**Comment by smblogger** [comment]
Felt a bit more stable of a build but dying on the following two patches. This is my primary station I'm working on so could be something kicking them out. I looked through the .rej files and it looks to me like everything is patched, but then they both fail anyways.  I will probably spin up a machine with only these required dev tools on it to eliminate possibilities.  When I comment out the ftp-support-thorium it does go on and makes it to the add-hevc-ffmpeg-decoder-parser.

Applied patch third_party/blink/renderer/platform/loader/ftp_directory_listing_test.cc cleanly.
setup.py: Failed git apply --reject ftp-support-thorium.patch

Applying patch chromium/config/Chromium/win/x64/libavcodec/parser_list.c with 1 reject...
Rejected hunk #1.
Applying patch ffmpeg_generated.gni with 5 rejects...
Rejected hunk #1.
Rejected hunk #2.
Rejected hunk #3.
Rejected hunk #4.
Rejected hunk #5.
setup.py: Failed git apply --reject add-hevc-ffmpeg-decoder-parser.patch

**Comment by gz83** [comment]
After local testing, no patches were found that failed to apply properly. It is recommended to check whether the local code files have been messed up.

**Comment by smblogger** [comment]
I have full deleted, src and thorium, synced, and retried a few times. Will let you know when I get it all spun up on a new machine.  I have so many dev languages on this one who knows what could be getting in the way.  I could say codec issues, but then that doesn't explain ftp.  Will invest some more time and get back to you.

**Comment by smblogger** [comment]
Second machine didn't have the fail on FTP oddly.  Still dies on the add-hevc-ffmpeg-decoder-parser patch.  Let me know if you see something I may have missed.  Notes for second machine install below.

https://github.com/Alex313031/Thorium-Win
Following instructions at https://github.com/Alex313031/thorium/blob/main/docs/BUILDING_WIN.md

c:\user\wil\Downloads\VisualStudioSetup.exe --add Microsoft.VisualStudio.Workload.NativeDesktop  --add Microsoft.VisualStudio.Component.VC.ATLMFC --add Microsoft.VisualStudio.Component.VC.Tools.ARM64 --add Microsoft.VisualStudio.Component.VC.MFC.ARM64 --includeRecommended

Installed Win 11 SDK from https://developer.microsoft.com/en-us/windows/downloads/sdk-archive/

Installed Git for windows

Installed depot_tools extracting in place within C:\src\depot_tools.

Installed python 3.10 but moved it down in the path. Enabled long paths at the end of install
c:\ python3 -V
Python 3.11.8  ←running from depot_tools properly

Mkdir c:\src
Cd src
Mkdir chromium
Cd chromium
Fetch chromium

git config --global user.name "BigWil"
git config --global user.email "bigwil@chromium.org"
git config --global core.autocrlf false
git config --global core.filemode false
git config --global branch.autosetuprebase always
git config --global core.longpaths true

Environment variables set:
DEPOT_TOOLS_WIN_TOOLCHAIN 0
NINJA_SUMMARIZE_BUILD 1
NUMBER_OF_PROCESSORS 8
vs2022_install  C:\Program Files\Microsoft Visual Studio\2022\Community

Cd c:\Users\wil
git clone --recursive https://github.com/Alex313031/thorium.git
Cd thorium
Python3 win_scripts\trunk.py
Python3 win_scripts\version.py

Everything runs fine so far…………..

Python3 win_scripts\setup.py

–snipped–
Applied patch chromium/config/Chromium/win/x64/libavcodec/codec_list.c cleanly.
Applied patch chromium/config/Chromium/win/x64/libavcodec/parser_list.c cleanly.
Applying patch ffmpeg_generated.gni with 5 rejects...
Rejected hunk #1.
Rejected hunk #2.
Rejected hunk #3.
Rejected hunk #4.
Rejected hunk #5.
Applied patch libavcodec/aarch64/autorename_libavcodec_aarch64_hevcdsp_idct_neon.S cleanly.
Applied patch libavcodec/aarch64/autorename_libavcodec_aarch64_hevcdsp_sao_neon.S cleanly.
Applied patch libavcodec/autorename_libavcodec_bswapdsp.c cleanly.
win_scripts\setup.py: Failed git apply --reject add-hevc-ffmpeg-decoder-parser.patch

**Comment by gz83** [comment]
Thank you for providing more information. I would like to confirm whether you have modified the corresponding python file on your local device according to my draft PR?

https://github.com/Alex313031/thorium/pull/886

**Comment by smblogger** [comment]
Yes. I manually changed them all already after your comment above.  Verified they were all still in place after trunk and version, and they are.  One of the two machines stopped complaining about dd-hevc-ffmpeg-decoder-parser.patch another weird anomaly.  But the win_scripts\win_build.py still fails saying it can't find build.ninja so obviously it still isn't prepping the build correctly.  Just not erroring in the version.py.  
ninja -d stats -C out/thorium thorium_all -j8
ninja: error: loading 'build.ninja': The system cannot find the file specified.

Machine 2 is still stuck at 
win_scripts\setup.py: Failed git apply --reject add-hevc-ffmpeg-decoder-parser.patch

Who would have thought a build would be so difficult.  I can build almost anything on a Linux box. Maybe I should try your Win Cross Build and see if that goes better.  Maybe these old basic x64 machines are causing the problem, not sure why that would be though.

**Comment by gz83** [comment]
Emm, I really can't reproduce the problem you encountered locally, you may still need to check your local code.

**Comment by smblogger** [comment]
Which "local code" are you recommending I review?  I spent a week reviewing and rebuilding. Any files you would recommend I compare that would effect the dd-hevc-ffmpeg-decoder-parser.patch ?

**Comment by gz83** [comment]
Local code refers to the Chromium code and Thorium code you pull locally.

In addition, I still recommend that you carefully read the documents I provided you earlier.

**Comment by smblogger** [comment]
Thanks. I will review all.  Just to help eliminate the possibility there is something in my machine specifically causing it, I pulled down a fresh copy of chromium and built their way from 
https://chromium.googlesource.com/chromium/src/+/main/docs/windows_build_instructions.md
gn gen out\Default --args="is_component_build = true is_debug = true blink_symbol_level = 0 v8_symbol_level = 0"
autoninja -C out\Default chrome mini_installer

Worked perfectly all the way through and I have a chrome.exe and mini_installer.   So the machines aren't the issue, just something about that patch.  I will bring the solution back here when I find it.

**Comment by gz83** [comment]
These patches should not cause any problems, otherwise the files in the win_scripts directory currently need to be updated, and the new versions are not yet ready to compile.

**Comment by gz83** [comment]
A new version for windows will be released soon.
